### ISSUE TITLE
[BUG] Phantom Directory

### ISSUE THREAD

**Issue by c0fe** [nextcloud/android#2416]
### Actual behaviour
-Tell us what happens
A mysterious directory "0" appears after initializing file upload and disappears the moment after any interaction with the application.
### Expected behaviour
-Tell us what should happen
 This directory should not exist.
### Steps to reproduce
1. Go to folder
2. Select file upload

### Environment data
Android version:
8.1.0
Device model: 
Nexus 5x
Stock or customized system:
Stock
Nextcloud app version:
3.0.3
Nextcloud server version:
13
### Screenshots
https://i.imgur.com/RXpuTxT.png
https://i.imgur.com/4A4jKNr.png

**Comment by tobiasKaminsky** [comment]
/storage/emulated/0/ is your sdcard/primary storage.
On android the storage system/structure is quite confusing.
Nextcloud open up on sdcard and therefore "0" is shown. Once you enter a subfolder this folder name is shown.

**Comment by c0fe** [comment]
@tobiasKaminsky yeah that is a bit confusing. though I do not have any SD Cards installed onto the phone. Anyway to make this appear easier to end users?

**Comment by tobiasKaminsky** [comment]
Fer first storage is always named this way, independent if it is sd card or or internal storage.
As this is the name of the folder we cannot do that much. Some other apps try to detect this and replace "0" by "sdcard", which can also lead to confusion and is only working for known storage paths...

**Comment by tobiasKaminsky** [comment]
Maybe @nextcloud/designers have an idea?

**Comment by jancborchardt** [comment]
We had this for the storage directory setting as well, where I brought up the same point. ;)

It is mighty confusing for anyone but programmers. We should do our best to try and "translate"/detect these known paths. Storage is after all our core competence so this is really important for flawless UX. :)

**Comment by joshtrichards** [comment]
Related: #1088
